extensions/pkg/controller/backupentry/genericactuator/actuator.go

```diff
@@ -54,33 +59,80 @@ func (a *actuator) deployEtcdBackupSecret(ctx context.Context, log logr.Logger,
 			log.Info("SeedNamespace for shoot not found. Avoiding etcd backup secret deployment")
 			return nil
 		}
-		log.Error(err, "Failed to get seed namespace")
-		return err
+		return fmt.Errorf("failed to get seed namespace: %w", err)
 	}
 	if namespace.DeletionTimestamp != nil {
 		log.Info("SeedNamespace for shoot is being terminated. Avoiding etcd backup secret deployment")
 		return nil
 	}
 
-	backupSecret, err := kubernetesutils.GetSecretByReference(ctx, a.client, &be.Spec.SecretRef)
+	backupEntrySecret, err := kubernetesutils.GetSecretByReference(ctx, a.client, &be.Spec.SecretRef)
 	if err != nil {
 		log.Error(err, "Failed to read backup extension secret")
 		return err
 	}
 
-	backupSecretData := backupSecret.DeepCopy().Data
-	if backupSecretData == nil {
-		backupSecretData = map[string][]byte{}
+	var (
+		backupEntrySecretData = backupEntrySecret.DeepCopy().Data
+		withWorkloadIdentity  = false
+	)
+
+	if purpose, ok := backupEntrySecret.Labels[securityv1alpha1constants.LabelPurpose]; ok && purpose == securityv1alpha1constants.LabelPurposeWorkloadIdentityTokenRequestor {
+		withWorkloadIdentity = true
+	}
```

Comment:
I think this is simpler to read.
```suggestion
	var (
		backupEntrySecretData = backupEntrySecret.DeepCopy().Data
		useWorkloadIdentity = backupEntrySecret.Labels[securityv1alpha1constants.LabelPurpose] == securityv1alpha1constants.LabelPurposeWorkloadIdentityTokenRequestor
	)
```

Reply:
Done with a5390c9350b745c138fb26b92cef96b9f1fc70c8